compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLKernelConfig.cpp

```diff
@@ -0,0 +1,407 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/LLVMGPU/ROCDLKernelConfig.h"
+
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "llvm/ADT/APInt.h"
+#include "llvm/ADT/STLExtras.h"
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Interfaces/FunctionInterfaces.h"
+
+namespace mlir::iree_compiler {
+
+namespace {
+
+using CodeGenPipeline = IREE::Codegen::DispatchLoweringPassPipeline;
+
+//===----------------------------------------------------------------------===//
+// Target Information
+//===----------------------------------------------------------------------===//
+
+struct TargetInfo {
+  bool hasWarpShuffle = false;
+  // These are listed in the order of preference, not necessarily monotonically.
+  SmallVector<int64_t, 2> supportedSubgroupSizes = {32};
+};
+
+static StringRef getTargetArch(mlir::FunctionOpInterface entryPoint) {
+  if (auto variantOp =
+          entryPoint->getParentOfType<IREE::HAL::ExecutableVariantOp>()) {
+    IREE::HAL::ExecutableTargetAttr targetAttr = variantOp.getTarget();
```

Comment:
nit: use `IREE::HAL::ExecutableTargetAttr::lookup(entryPoint)`, no need to search for the variant parent first anymore.